gimme

```diff
@@ -164,38 +169,53 @@ _extract() {
 	tar -xf "${1}" -C "${2}" --strip-components 1
 }
 
+# _setup_bootstrap
+_setup_bootstrap() {
+	local versions=("1.6" "1.5" "1.4")
+
+	# try existing
+	for v in "${versions[@]}" ; do
+		for candidate in "${GIMME_ENV_PREFIX}/go${v}"*".env" ; do
+			if [ -s "$candidate" ]; then
+				export GOROOT_BOOTSTRAP="$(source "${candidate}" 2>/dev/null && go env GOROOT)"
+				return 0
+			fi
+		done
+	done
+
+	# try binary
+	for v in "${versions[@]}" ; do
+		if [ -n  "$(_try_binary ${v} "${GIMME_HOSTARCH}")" ]; then
+			export GOROOT_BOOTSTRAP="${GIMME_VERSION_PREFIX}/go${v}.${GIMME_OS}.${GIMME_HOSTARCH}"
+			return 0
+		fi
+	done
+
+	echo >&2 "Unable to setup go bootstrap from existing or binary";
+	return 1;
+}
+
 # _compile "dir"
 _compile() {
 	(
-		if grep -q GOROOT_BOOTSTRAP "${1}/src/make.bash" &> /dev/null ; then
-			local bootstrap_root=""
-			local bootstrap_candidates=(
-				# we just need _a_ working Go 1.4+ build, really
-				"${GIMME_ENV_PREFIX}/go1.6"*".env"
-				"${GIMME_ENV_PREFIX}/go1.5"*".env"
-				"${GIMME_ENV_PREFIX}/go1.4"*".env"
-			)
-			for candidate in "${bootstrap_candidates[@]}" ; do
-				if [ -s "$candidate" ]; then
-					bootstrap_root="$candidate"
-					break
-				fi
-			done
-			if [ -z "${bootstrap_root}" ] ; then
-				echo >&2 "error: Missing bootstrap installation"
-				echo >&2 "error: Unable to compile go from git clone"
-				return 1
-			fi
-			export GOROOT_BOOTSTRAP="$(source "${bootstrap_root}" 2>/dev/null && go env GOROOT)"
+		if grep -q GOROOT_BOOTSTRAP "${1}/src/make.bash" &> /dev/null; then
+			_setup_bootstrap || return 1
 		fi
 		cd "${1}"
 		if [[ -d .git ]] ; then
 			git clean -dfx -q
 		fi
 		cd src
 		export GOOS="${GIMME_OS}" GOARCH="${GIMME_ARCH}"
-		local make_log="${1}/make.${GOOS}.${GOARCH}.log"
-		./make.bash &> $make_log || return 1
+		export CGO_ENABLED="${GIMME_CGO_ENABLED}"
+		export CC_FOR_TARGET="${GIMME_CC_FOR_TARGET}"
+
+		if [ "${GIMME_DEBUG}" = "2" ]; then
+			./make.bash 1>&2 || return 1
```

Comment:
Can this `tee $make_log` instead of only going to stdout?  That way the log is present for any value of `GIMME_DEBUG`.